skbio/metadata/_mixin.py

```diff
@@ -368,7 +371,15 @@ def __deepcopy__(self, memo):
 
     def _deepcopy_(self, memo):
         if self.has_positional_metadata():
-            return copy.deepcopy(self.positional_metadata, memo)
+            # `copy.deepcopy` no longer recursively copies contents of the
+            # DataFrame, so we must handle the deep copy ourselves.
+            # Reference: https://github.com/pandas-dev/pandas/issues/17406
+            df = self.positional_metadata
+            data_cp = copy.deepcopy(df.values.tolist(), memo)
+            return pd.DataFrame(data_cp,
+                                index=df.index.copy(deep=True),
+                                columns=df.columns.copy(deep=True),
```

Comment:
These are made of hashable objects right? So even a shallow copy is deep enough for our purposes?

Reply:
They're documented to only contain hashable objects but in practice you can create `Index` objects with non-hashables, so this seemed safer to do.